modules/storage/mysql-single-server/main.bicep

```diff
@@ -0,0 +1,212 @@
+param administratorLogin string
+
+@secure()
+@description('Required. The administrator password of the server. Can only be specified when the server is being created.')
+param administratorLoginPassword string
+
+@description('Optional. The number of days a backup is retained.')
+@minValue(7)
+@maxValue(35)
+param backupRetentionDays int = 35
+
+@description('Optional. The mode to create a new server.')
+@allowed([
+  'Default'
+  'GeoRestore'
+  'PointInTimeRestore'
+  'Replica'
+])
+param createMode string = 'Default'
+
+@description('Optional. List of databases to create on server.')
+param databases array = []
+
+@description('Optional. List of server configurations to create on server.')
+param serverConfigurations array = []
+
+@description('Optional. Status showing whether the server enabled infrastructure encryption..')
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+param infrastructureEncryption string = 'Disabled'
+
+@description('Optional. List of firewall rules to create on server.')
+param firewallRules array = []
+
+@description('Optional. List of privateEndpoints to create on mysql server.')
+param privateEndpoints array = []
+
+@description('Optional. Enable or disable geo-redundant backups.')
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+param geoRedundantBackup string = 'Enabled'
+
+@description('Optional. Enforce a minimal Tls version for the server.')
+@allowed([
+  'TLS1_0'
+  'TLS1_1'
+  'TLS1_2'
+  'TLSEnforcementDisabled'
+])
+param minimalTlsVersion string = 'TLS1_2'
+
+@description('Optional. Restore point creation time (ISO8601 format), specifying the time to restore from.')
+param restorePointInTime string = ''
+
+@description('Optional. Whether or not public network access is allowed for this server.')
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+param publicNetworkAccess string = 'Disabled'
+
+@description('Required. The name of the server.')
+param serverName string
+
+@description('Optional.	The name of the sku, typically, tier + family + cores, e.g. B_Gen4_1, GP_Gen5_8.')
+param skuName string = 'GP_Gen5_2'
+
+@description('Optional. The source server resource id to restore from, e.g. "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rg1/providers/Microsoft.DBforMySQL/flexibleServers/server1" . It\'s required when "createMode" is "GeoRestore" or "Replica" or "PointInTimeRestore".')
+param sourceServerResourceId string = ''
+
+@description('Optional. Enable ssl enforcement or not when connect to server.')
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+param sslEnforcement string = 'Enabled'
+
+@description('Optional. Auto grow of storage.')
+param storageAutogrow bool = true
+
+@description('Optional. The storage size of the server.')
+param storageSizeGB int = 32
+
+@description('Optional. The version of the MySQL server.')
+@allowed([
+  '5.6'
+  '5.7'
+  '8.0'
+])
+param version string = '8.0'
```

Comment:
```suggestion
@allowed([ '5.6', '5.7', '8.0' ])
param version string = '8.0'
```

Reply:
nitpick, but I like using singleline when the line is less than about 100 characters.